suites/tipset/message_dedup.go

```diff
@@ -92,4 +92,25 @@ func TestBlockMessageDeduplication(t *testing.T, factory state.Factories) {
 		td.AssertBalance(receiver, big_spec.NewInt(100))
 	})
 
+	t.Run("apply duplicate BLS and SECP messages", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+		tipB := drivers.NewTipSetMessageBuilder(td)
+		blkB := drivers.NewBlockBuilder(td, td.ExeCtx.Miner)
+
+		senderInitialBal := big_spec.NewInt(10_000_000)
+		_, senderID := td.NewAccountActor(address.SECP256K1, senderInitialBal)
+		_, receiverID := td.NewAccountActor(address.SECP256K1, big_spec.Zero())
+
+		amountSent := big_spec.NewInt(100)
+		result := tipB.WithBlockBuilder(
+			// using ID addresses will ensure the BLS message and the unsigned message encapsulated in the SECP message
+			// have the same CID.
+			blkB.WithBLSMessageOk(td.MessageProducer.Transfer(receiverID, senderID, chain.Nonce(0), chain.Value(amountSent))).
+				WithSECPMessageDropped(td.MessageProducer.Transfer(receiverID, senderID, chain.Nonce(0), chain.Value(amountSent))),
+		).ApplyAndValidate()
```

Comment:
Show that there is only one receipt

Reply:
`WithSECPMessageDropped` implies this, but for cality I have added a check below.